File: androidApp/src/main/kotlin/com/torve/android/ui/settings/DiagnosticsScreen.kt
package com.torve.android.ui.settings

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.os.Build
import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.torve.android.R
import com.torve.android.ui.components.BackButton
import com.torve.android.ui.theme.Amber
import com.torve.android.ui.theme.Charcoal
import com.torve.android.ui.theme.Obsidian
import com.torve.android.ui.theme.Silver
import com.torve.android.ui.theme.Snow
import com.torve.presentation.settings.SettingsUiState
import com.torve.presentation.settings.SettingsViewModel
import org.koin.compose.koinInject

@Composable
fun DiagnosticsScreen(
    onBack: () -> Unit,
    viewModel: SettingsViewModel = koinInject(),
) {
    val state by viewModel.state.collectAsState()
    val context = LocalContext.current
    val appVersion = runCatching {
        context.packageManager.getPackageInfo(context.packageName, 0).versionName ?: "unknown"
    }.getOrElse { "unknown" }
    val diagnostics = buildDiagnosticsText(
        appVersion = appVersion,
        buildType = "release",
        deviceModel = "${Build.MANUFACTURER} ${Build.MODEL}",
        androidVersion = "${Build.VERSION.RELEASE} (SDK ${Build.VERSION.SDK_INT})",
        settingsState = state,
    )

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Obsidian)
            .statusBarsPadding()
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
    ) {
        BackButton(onClick = onBack)
        Spacer(Modifier.height(8.dp))
        Text(stringResource(R.string.diagnostics_title), style = MaterialTheme.typography.titleLarge, color = Snow, fontWeight = FontWeight.Bold)
        Spacer(Modifier.height(12.dp))

        // Device & App info
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text("App: $appVersion", color = Snow, style = MaterialTheme.typography.bodyMedium)
                Text("Build: release", color = Silver, style = MaterialTheme.typography.bodySmall)
                Text("Device: ${Build.MANUFACTURER} ${Build.MODEL}", color = Silver, style = MaterialTheme.typography.bodySmall)
                Text("Android: ${Build.VERSION.RELEASE} (SDK ${Build.VERSION.SDK_INT})", color = Silver, style = MaterialTheme.typography.bodySmall)
                Text("Region: ${state.regionCode}", color = Silver, style = MaterialTheme.typography.bodySmall)
            }
        }
        Spacer(Modifier.height(12.dp))

        // Service Health
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_service_health), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                StatusRow("Cloud", state.debridConnected)
                StatusRow("Trakt", state.traktConnected)
                StatusRow("SIMKL", state.simklConnected)
                StatusRow("Jellyfin", state.jellyfinApiKey.isNotBlank())
                StatusRow("Plex", state.plexConnected)
                StatusRow("AI (${state.aiProvider.name})", state.activeAiApiKey.isNotBlank())
                StatusRow("MDBList", state.mdblistApiKey.isNotBlank())
            }
        }
        Spacer(Modifier.height(12.dp))

        // Sync timestamps
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_sync_status), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                Text("Trakt last sync: ${formatTimestamp(state.traktLastSyncTime)}", color = Silver, style = MaterialTheme.typography.bodySmall)
                Text("Availability last sync: ${formatTimestamp(state.availabilityLastSyncTime)}", color = Silver, style = MaterialTheme.typography.bodySmall)
                Text("Library overlay last sync: ${formatTimestamp(state.libraryOverlayLastSyncTime)}", color = Silver, style = MaterialTheme.typography.bodySmall)
            }
        }
        Spacer(Modifier.height(12.dp))

        // Content & Configuration
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_content_config), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                DiagRow("Card style presets", "${state.cardStylePresets.size}")
                DiagRow("Global default preset", state.globalDefaultPresetId ?: "default")
                DiagRow("Regex patterns", "${state.regexPatterns.size} active")
                DiagRow("Stream groups", "${state.streamGroups.size}")
                DiagRow("Kodi devices", "${state.kodiHosts.size}")
                DiagRow("AI provider", state.aiProvider.name)
                DiagRow("AI key configured", if (state.activeAiApiKey.isNotBlank()) "Yes" else "No")
                DiagRow("Cloud provider", state.debridProvider.name)
                DiagRow("Connected cloud services", "${state.connectedDebridProviders.size}")
            }
        }
        Spacer(Modifier.height(12.dp))

        // Stream Preferences
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_stream_prefs), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                DiagRow("Max quality", state.maxQuality.name)
                DiagRow("Min quality", state.minQuality.name)
                DiagRow("Max file size", state.maxFileSizeMb?.let { "${it} MB" } ?: "No limit")
                DiagRow("Cached only", state.cachedOnly.toString())
                DiagRow("Codec preference", state.codecPreference.name)
                DiagRow("HDR mode", state.hdrMode.name)
                DiagRow("Dedupe results", state.dedupeResults.toString())
            }
        }
        Spacer(Modifier.height(12.dp))

        // Ratings
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_rating_config), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                DiagRow("Enabled providers", state.ratingPrefs.enabledProviders.joinToString(", ") { it.name })
                DiagRow("Pill style", state.ratingPrefs.pillStyle.name)
                DiagRow("Pill position", state.ratingPrefs.pillPosition.displayName)
                DiagRow("Max ratings on card", "${state.ratingPrefs.maxRatingsOnCard}")
                DiagRow("Show on detail page", state.ratingPrefs.showRatingsOnDetailPage.toString())
                DiagRow("Torve on cards", state.ratingPrefs.showTorveScoreOnCards.toString())
                DiagRow("Landscape ratings", state.ratingPrefs.allowRatingsOnLandscapeCards.toString())
            }
        }
        Spacer(Modifier.height(12.dp))

        // Integrations detail
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Column(modifier = Modifier.padding(14.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(stringResource(R.string.diagnostics_integrations_detail), color = Snow, fontWeight = FontWeight.SemiBold, style = MaterialTheme.typography.bodyMedium)
                DiagRow("Jellyfin URL", if (state.jellyfinServerUrl.isNotBlank()) state.jellyfinServerUrl else "Not configured")
                DiagRow("Jellyfin profiles", "${state.jellyfinProfiles.size}")
                DiagRow("Plex URL", if (state.plexServerUrl.isNotBlank()) state.plexServerUrl else "Not configured")
                DiagRow("Plex connected", state.plexConnected.toString())
                DiagRow("Trakt scrobble", state.traktScrobbleEnabled.toString())
                DiagRow("Theme", state.themeMode.name)
                DiagRow("Language", state.appLanguage.displayName)
            }
        }
        Spacer(Modifier.height(12.dp))

        // Raw diagnostics
        Card(modifier = Modifier.fillMaxWidth(), colors = CardDefaults.cardColors(containerColor = Charcoal)) {
            Text(
                diagnostics,
                modifier = Modifier.padding(12.dp),
                color = Silver,
                style = MaterialTheme.typography.bodySmall,
            )
        }
        Spacer(Modifier.height(12.dp))

        // Actions
        val copiedMessage = stringResource(R.string.diagnostics_copied)
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.fillMaxWidth()) {
            Button(
                onClick = {
                    val intent = Intent(Intent.ACTION_SEND).apply {
                        type = "text/plain"
                        putExtra(Intent.EXTRA_SUBJECT, "Torve diagnostics")
                        putExtra(Intent.EXTRA_TEXT, diagnostics)
                    }
                    context.startActivity(Intent.createChooser(intent, "Share diagnostics"))
                },
                modifier = Modifier.weight(1f),
                colors = ButtonDefaults.buttonColors(containerColor = Amber, contentColor = Obsidian),
            ) {
                Text(stringResource(R.string.common_share))
            }
            OutlinedButton(
                onClick = {
                    val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
                    clipboard.setPrimaryClip(ClipData.newPlainText("Torve diagnostics", diagnostics))
                    Toast.makeText(context, copiedMessage, Toast.LENGTH_SHORT).show()
                },
                modifier = Modifier.weight(1f),
                colors = ButtonDefaults.outlinedButtonColors(contentColor = Amber),
            ) {
                Text(stringResource(R.string.diagnostics_copy))
            }
        }
        Spacer(Modifier.height(16.dp))
    }
}

@Composable
private fun StatusRow(label: String, connected: Boolean) {
    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
        Text(label, color = Silver, style = MaterialTheme.typography.bodySmall)
        Text(
            if (connected) stringResource(R.string.diagnostics_connected) else stringResource(R.string.diagnostics_not_configured),
            color = if (connected) Snow else Silver,
            style = MaterialTheme.typography.bodySmall,
            fontWeight = if (connected) FontWeight.SemiBold else FontWeight.Normal,
        )
    }
}

@Composable
private fun DiagRow(label: String, value: String) {
    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
        Text(label, color = Silver, style = MaterialTheme.typography.bodySmall, modifier = Modifier.weight(1f))
        Text(value, color = Snow, style = MaterialTheme.typography.bodySmall, fontWeight = FontWeight.Medium)
    }
}

private fun formatTimestamp(epochMs: Long?): String {
    if (epochMs == null) return "Never"
    return java.text.SimpleDateFormat("MMM d, HH:mm", java.util.Locale.getDefault())
        .format(java.util.Date(epochMs))
}

private fun buildDiagnosticsText(
    appVersion: String,
    buildType: String,
    deviceModel: String,
    androidVersion: String,
    settingsState: SettingsUiState,
): String {
    val redacted = mapOf(
        "traktConnected" to settingsState.traktConnected.toString(),
        "debridConnected" to settingsState.debridConnected.toString(),
        "debridProvider" to settingsState.debridProvider.name,
        "connectedDebridCount" to settingsState.connectedDebridProviders.size.toString(),
        "simklConnected" to settingsState.simklConnected.toString(),
        "jellyfinConfigured" to settingsState.jellyfinApiKey.isNotBlank().toString(),
        "jellyfinUrl" to settingsState.jellyfinServerUrl.takeIf { it.isNotBlank() }.orEmpty(),
        "plexConnected" to settingsState.plexConnected.toString(),
        "plexUrl" to settingsState.plexServerUrl.takeIf { it.isNotBlank() }.orEmpty(),
        "regionCode" to settingsState.regionCode,
        "traktLastSync" to (settingsState.traktLastSyncTime?.toString() ?: ""),
        "availabilityLastSync" to (settingsState.availabilityLastSyncTime?.toString() ?: ""),
        "libraryOverlayLastSync" to (settingsState.libraryOverlayLastSyncTime?.toString() ?: ""),
        "ratingProviders" to settingsState.ratingPrefs.enabledProviders.joinToString(",") { it.name },
        "ratingPillStyle" to settingsState.ratingPrefs.pillStyle.name,
        "ratingPillPosition" to settingsState.ratingPrefs.pillPosition.name,
        "maxRatingsOnCard" to settingsState.ratingPrefs.maxRatingsOnCard.toString(),
        "cardStylePresets" to settingsState.cardStylePresets.size.toString(),
        "globalDefaultPreset" to (settingsState.globalDefaultPresetId ?: "default"),
        "regexPatterns" to settingsState.regexPatterns.size.toString(),
        "streamGroups" to settingsState.streamGroups.size.toString(),
        "kodiHosts" to settingsState.kodiHosts.size.toString(),
        "aiProvider" to settingsState.aiProvider.name,
        "aiKeyConfigured" to settingsState.activeAiApiKey.isNotBlank().toString(),
        "mdblistConfigured" to settingsState.mdblistApiKey.isNotBlank().toString(),
        "maxQuality" to settingsState.maxQuality.name,
        "minQuality" to settingsState.minQuality.name,
        "maxFileSizeMb" to (settingsState.maxFileSizeMb?.toString() ?: "unlimited"),
        "cachedOnly" to settingsState.cachedOnly.toString(),
        "codecPreference" to settingsState.codecPreference.name,
        "hdrMode" to settingsState.hdrMode.name,
        "traktScrobble" to settingsState.traktScrobbleEnabled.toString(),
        "dedupeResults" to settingsState.dedupeResults.toString(),
        "theme" to settingsState.themeMode.name,
        "language" to settingsState.appLanguage.code,
    )
    return buildString {
        appendLine("Torve diagnostics")
        appendLine("appVersion=$appVersion")
        appendLine("buildType=$buildType")
        appendLine("device=$deviceModel")
        appendLine("android=$androidVersion")
        appendLine("settings=${redacted.entries.joinToString(";") { "${it.key}=${it.value}" }}")
    }
}

File: iosApp/iosApp/Views/Settings/DiagnosticsScreen.swift
import SwiftUI
import shared

struct DiagnosticsScreen: View {
    @StateObject private var wrapper = SettingsViewModelWrapper()

    @State private var showCopiedToast = false

    private var appVersion: String {
        Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "?"
    }

    private var buildNumber: String {
        Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? "?"
    }

    private var deviceModel: String {
        var systemInfo = utsname()
        uname(&systemInfo)
        let machine = withUnsafePointer(to: &systemInfo.machine) {
            $0.withMemoryRebound(to: CChar.self, capacity: 1) {
                String(cString: $0)
            }
        }
        return machine
    }

    private var systemVersion: String {
        "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
    }

    private var diagnosticsText: String {
        let s = wrapper.state
        let aiKeyConfigured = !s.activeAiApiKey.isEmpty
        let jellyfinConfigured = !s.jellyfinApiKey.isEmpty
        let mdblistConfigured = !s.mdblistApiKey.isEmpty

        let settings: [(String, String)] = [
            ("traktConnected", "\(s.traktConnected)"),
            ("debridConnected", "\(s.debridConnected)"),
            ("debridProvider", s.debridProvider.name),
            ("connectedDebridCount", "\(s.connectedDebridProviders.count)"),
            ("simklConnected", "\(s.simklConnected)"),
            ("jellyfinConfigured", "\(jellyfinConfigured)"),
            ("jellyfinUrl", s.jellyfinServerUrl.isEmpty ? "" : s.jellyfinServerUrl),
            ("plexConnected", "\(s.plexConnected)"),
            ("plexUrl", s.plexServerUrl.isEmpty ? "" : s.plexServerUrl),
            ("regionCode", s.regionCode),
            ("traktLastSync", s.traktLastSyncTime.map { "\($0.int64Value)" } ?? ""),
            ("availabilityLastSync", s.availabilityLastSyncTime.map { "\($0.int64Value)" } ?? ""),
            ("libraryOverlayLastSync", s.libraryOverlayLastSyncTime.map { "\($0.int64Value)" } ?? ""),
            ("ratingProviders", (s.ratingPrefs.enabledProviders as? [AnyObject] ?? []).map { "\($0)" }.joined(separator: ",")),
            ("ratingPillStyle", s.ratingPrefs.pillStyle.name),
            ("ratingPillPosition", s.ratingPrefs.pillPosition.name),
            ("maxRatingsOnCard", "\(s.ratingPrefs.maxRatingsOnCard)"),
            ("cardStylePresets", "\(s.cardStylePresets.count)"),
            ("globalDefaultPreset", s.globalDefaultPresetId ?? "default"),
            ("regexPatterns", "\(s.regexPatterns.count)"),
            ("streamGroups", "\(s.streamGroups.count)"),
            ("kodiHosts", "\(s.kodiHosts.count)"),
            ("aiProvider", s.aiProvider.name),
            ("aiKeyConfigured", "\(aiKeyConfigured)"),
            ("mdblistConfigured", "\(mdblistConfigured)"),
            ("maxQuality", s.maxQuality.name),
            ("minQuality", s.minQuality.name),
            ("maxFileSizeMb", s.maxFileSizeMb.map { "\($0.int32Value)" } ?? "unlimited"),
            ("cachedOnly", "\(s.cachedOnly)"),
            ("codecPreference", s.codecPreference.name),
            ("hdrMode", s.hdrMode.name),
            ("traktScrobble", "\(s.traktScrobbleEnabled)"),
            ("dedupeResults", "\(s.dedupeResults)"),
            ("theme", s.themeMode.name),
            ("language", s.appLanguage.code),
        ]
        let settingsLine = settings.map { "\($0.0)=\($0.1)" }.joined(separator: ";")
        return """
        Torve diagnostics
        appVersion=\(appVersion)
        buildType=release
        device=\(deviceModel)
        system=\(systemVersion)
        settings=\(settingsLine)
        """
    }

    var body: some View {
        List {
            // 1. Device & App Info
            Section {
                diagRow("App", value: appVersion)
                diagRow("Build", value: buildNumber)
                diagRow("Device", value: deviceModel)
                diagRow("System", value: systemVersion)
                diagRow("Region", value: wrapper.state.regionCode)
            } header: {
                Text("Device & App Info")
            }

            // 2. Service Health
            Section {
                statusRow("Cloud", connected: wrapper.state.debridConnected)
                statusRow("Trakt", connected: wrapper.state.traktConnected)
                statusRow("SIMKL", connected: wrapper.state.simklConnected)
                statusRow("Jellyfin", connected: !wrapper.state.jellyfinApiKey.isEmpty)
                statusRow("Plex", connected: wrapper.state.plexConnected)
                statusRow("AI (\(wrapper.state.aiProvider.name))", connected: !wrapper.state.activeAiApiKey.isEmpty)
                statusRow("MDBList", connected: !wrapper.state.mdblistApiKey.isEmpty)
            } header: {
                Text("Service Health")
            }

            // 3. Sync Timestamps
            Section {
                diagRow("Trakt last sync", value: formatOptionalTimestamp(wrapper.state.traktLastSyncTime))
                diagRow("Availability last sync", value: formatOptionalTimestamp(wrapper.state.availabilityLastSyncTime))
                diagRow("Library overlay last sync", value: formatOptionalTimestamp(wrapper.state.libraryOverlayLastSyncTime))
            } header: {
                Text("Sync Status")
            }

            // 4. Content & Configuration
            Section {
                diagRow("Card style presets", value: "\(wrapper.state.cardStylePresets.count)")
                diagRow("Global default preset", value: wrapper.state.globalDefaultPresetId ?? "default")
                diagRow("Regex patterns", value: "\(wrapper.state.regexPatterns.count) active")
                diagRow("Stream groups", value: "\(wrapper.state.streamGroups.count)")
                diagRow("Kodi devices", value: "\(wrapper.state.kodiHosts.count)")
                diagRow("AI provider", value: wrapper.state.aiProvider.name)
                diagRow("AI key configured", value: wrapper.state.activeAiApiKey.isEmpty ? "No" : "Yes")
                diagRow("Cloud provider", value: wrapper.state.debridProvider.name)
                diagRow("Connected cloud services", value: "\(wrapper.state.connectedDebridProviders.count)")
            } header: {
                Text("Content & Configuration")
            }

            // 5. Stream Preferences
            Section {
                diagRow("Max quality", value: wrapper.state.maxQuality.name)
                diagRow("Min quality", value: wrapper.state.minQuality.name)
                if let maxSize = wrapper.state.maxFileSizeMb?.int32Value {
                    diagRow("Max file size", value: "\(maxSize) MB")
                } else {
                    diagRow("Max file size", value: "No limit")
                }
                diagRow("Cached only", value: "\(wrapper.state.cachedOnly)")
                diagRow("Codec preference", value: wrapper.state.codecPreference.name)
                diagRow("HDR mode", value: wrapper.state.hdrMode.name)
                diagRow("Dedupe results", value: "\(wrapper.state.dedupeResults)")
            } header: {
                Text("Stream Preferences")
            }

            // 6. Ratings Configuration
            Section {
                let prefs = wrapper.state.ratingPrefs
                let providers = (prefs.enabledProviders as? [AnyObject] ?? []).map { "\($0)" }.joined(separator: ", ")
                diagRow("Enabled providers", value: providers.isEmpty ? "None" : providers)
                diagRow("Pill style", value: prefs.pillStyle.name)
                diagRow("Pill position", value: prefs.pillPosition.displayName)
                diagRow("Max ratings on card", value: "\(prefs.maxRatingsOnCard)")
                diagRow("Show on detail page", value: "\(prefs.showRatingsOnDetailPage)")
                diagRow("Torve on cards", value: "\(prefs.showTorveScoreOnCards)")
                diagRow("Landscape ratings", value: "\(prefs.allowRatingsOnLandscapeCards)")
            } header: {
                Text("Rating Configuration")
            }

            // 7. Integrations Detail
            Section {
                diagRow("Jellyfin URL", value: wrapper.state.jellyfinServerUrl.isEmpty ? "Not configured" : wrapper.state.jellyfinServerUrl)
                diagRow("Jellyfin profiles", value: "\(wrapper.state.jellyfinProfiles.count)")
                diagRow("Plex URL", value: wrapper.state.plexServerUrl.isEmpty ? "Not configured" : wrapper.state.plexServerUrl)
                diagRow("Plex connected", value: "\(wrapper.state.plexConnected)")
                diagRow("Trakt scrobble", value: "\(wrapper.state.traktScrobbleEnabled)")
                diagRow("Theme", value: wrapper.state.themeMode.name)
                diagRow("Language", value: wrapper.state.appLanguage.displayName)
            } header: {
                Text("Integrations Detail")
            }

            // 8. Raw Diagnostics
            Section {
                Text(diagnosticsText)
                    .font(.caption2)
                    .foregroundColor(SVColor.onSurfaceVariant)
                    .textSelection(.enabled)
            } header: {
                Text("Raw Diagnostics")
            }

            // 9. Actions
            Section {
                Button {
                    shareDiagnostics()
                } label: {
                    Label("Share Diagnostics", systemImage: "square.and.arrow.up")
                        .foregroundColor(SVColor.amber)
                }

                Button {
                    UIPasteboard.general.string = diagnosticsText
                    showCopiedToast = true
                    DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                        showCopiedToast = false
                    }
                } label: {
                    Label("Copy Diagnostics", systemImage: "doc.on.doc")
                        .foregroundColor(SVColor.amber)
                }
            }
        }
        .listStyle(.insetGrouped)
        .navigationTitle("Diagnostics")
        .overlay(alignment: .bottom) {
            if showCopiedToast {
                Text("Copied to clipboard")
                    .font(.subheadline)
                    .fontWeight(.medium)
                    .padding(.horizontal, 16)
                    .padding(.vertical, 10)
                    .background(.ultraThinMaterial)
                    .clipShape(Capsule())
                    .padding(.bottom, 24)
                    .transition(.move(edge: .bottom).combined(with: .opacity))
                    .animation(.easeInOut, value: showCopiedToast)
            }
        }
    }

    // MARK: - Row Helpers

    private func diagRow(_ label: String, value: String) -> some View {
        HStack {
            Text(label)
                .foregroundColor(SVColor.onSurfaceVariant)
            Spacer()
            Text(value)
                .foregroundColor(SVColor.onSurface)
                .fontWeight(.medium)
        }
        .font(.subheadline)
    }

    private func statusRow(_ label: String, connected: Bool) -> some View {
        HStack {
            Text(label)
                .font(.subheadline)
                .foregroundColor(SVColor.onSurfaceVariant)
            Spacer()
            Text(connected ? "Connected" : "Not configured")
                .font(.caption)
                .fontWeight(connected ? .semibold : .regular)
                .padding(.horizontal, 8)
                .padding(.vertical, 2)
                .background(connected ? SVColor.emerald.opacity(0.2) : Color.secondary.opacity(0.2))
                .foregroundColor(connected ? SVColor.emerald : .secondary)
                .clipShape(Capsule())
        }
    }

    // MARK: - Timestamp Formatting

    private func formatOptionalTimestamp(_ millis: KotlinLong?) -> String {
        guard let millis = millis else { return "Never" }
        return formatTimestamp(millis.int64Value)
    }

    private func formatTimestamp(_ millis: Int64) -> String {
        let date = Date(timeIntervalSince1970: TimeInterval(millis) / 1000.0)
        let formatter = DateFormatter()
        formatter.dateFormat = "MMM d, HH:mm"
        return formatter.string(from: date)
    }

    // MARK: - Actions

    private func shareDiagnostics() {
        let activityVC = UIActivityViewController(
            activityItems: [diagnosticsText],
            applicationActivities: nil
        )
        if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
           let rootVC = windowScene.windows.first?.rootViewController {
            // Handle iPad popover
            if let popover = activityVC.popoverPresentationController {
                popover.sourceView = rootVC.view
                popover.sourceRect = CGRect(
                    x: rootVC.view.bounds.midX,
                    y: rootVC.view.bounds.midY,
                    width: 0,
                    height: 0
                )
                popover.permittedArrowDirections = []
            }
            rootVC.present(activityVC, animated: true)
        }
    }
}
